src/Components/SmartComponents/CVEs/VulnerabilitiesCves.js

```diff
@@ -24,195 +21,122 @@ import {
+    const handleCveSelect = (isSelected, cveNames) => {
+        dispatch(selectCve(cveNames || []));
     };
 
-    selectCves = (isSelected, cveNames) => {
-        this.props.selectCve(cveNames || []);
+    const downloadReport = format => {
+        DownloadReport.exec(fetchCveListByAccount, parameters, format, 'cves');
     };
 
-    toggleCveDescription = (cveList) => {
-        const { parameters } = this.props;
-        const { expandCveDescription } = parameters;
-        const openedCves = expandCveDescription
-            ? cveList.data.filter(cve => cve.id).map(cve => cve.id)
-            : parameters.openedCves.length ? parameters.openedCves : [];
-
-        this.props.openCve(openedCves);
-    }
-
-    createUrlParams = allParams => {
-        const params = { ...allParams };
-        params.show_irrelevant = !params.show_all;
-        const allowedParams = [
-            'show_irrelevant',
-            'cvss_filter',
-            'publish_date',
-            'filter',
-            'limit',
-            'offset',
-            'page',
-            'page_size',
-            'sort',
-            'impact',
-            'business_risk_id',
-            'status_id',
-            'security_rule'
-        ];
-        Object.keys(allParams).forEach(
-            key =>
-                (params[key] === undefined || params[key] === '' || !allowedParams.includes(key) || params[key] === false) &&
-                delete params[key]
+    const showBusinessRiskModal = cvesList => {
+        const  { meta } = cves;
+        setBusinessRiskModal(() => () =>
+            (<BusinessRiskModal cves={cvesList} updateRef={() => updateRef(meta, apply)} />)
         );
-        const queryString = qs.stringify(params);
-        this.props.history.push('?' + queryString);
     };
 
-    sendRequest = () => {
-        const { fetchData, parameters } = this.props;
-        //TODO: need a better way of doing this
-        const showAllParam = Object.prototype.hasOwnProperty.call(parameters, 'show_all') && !parameters.show_all;
-
-        fetchData && fetchData({ ...parameters, show_all: showAllParam })
-        .then(response => {
-            const { value } = response;
-            if (parameters.expandCveDescription) {
-                this.toggleCveDescription(value);
-            }
-        });
-
-        this.createUrlParams(parameters);
-    };
-
-    downloadReport = format => {
-        const { parameters } = this.props;
-        const params = { ...parameters, show_all: !parameters.show_all };
-
-        DownloadReport.exec(fetchCveListByAccount, params, format, 'cves');
-    };
-
-    showBusinessRiskModal = cves => {
-        const { cveList: { meta } } = this.props;
-        this.setState({
-            BusinessRiskModal: () => <BusinessRiskModal cves={cves} updateRef={() => updateRef(meta, this.apply)} />
-        });
+    const showStatusModal = cvesList => {
+        const  { meta } = cves;
+        setStatusModal(() => () =>
```

Comment:
@mkholjuraev do you really need `() => () =>` ?

Reply:
yes, it is. as we have one more parent component. if we get rid of one component it is causing system to crash